src/components/input/inputFileAvatar/inputFileAvatar.tsx

```diff
@@ -0,0 +1,173 @@
+            const onImageLoad = () => {
+                const isBelowMinDimension =
+                    minDimension > 0 && (image.width < minDimension || image.height < minDimension);
+                const isAboveMaxDimension =
+                    maxDimension > 0 && (image.width > maxDimension || image.height > maxDimension);
+                if (onlySquare && image.height !== image.width) {
+                    onFileError?.(InputFileAvatarError.SQUARE_ONLY);
+                    setIsLoading(false);
+                    setCurrentState(InputFileAvatarState.ERROR);
+                } else if (isBelowMinDimension || isAboveMaxDimension) {
+                    onFileError?.(InputFileAvatarError.WRONG_DIMENSION);
+                    setIsLoading(false);
+                    setCurrentState(InputFileAvatarState.ERROR);
+                } else {
+                    setImagePreview(image.src);
+                    setIsLoading(false);
+                    setCurrentState(InputFileAvatarState.SUCCESS);
+                    onFileError?.(undefined);
+                    onFileSelect?.(file);
+                }
+            };
+
+            image.addEventListener('load', onImageLoad);
+            image.addEventListener('error', () => {
+                onFileError?.(InputFileAvatarError.UNKNOWN_ERROR);
+                setCurrentState(InputFileAvatarState.ERROR);
+            });
+            image.src = URL.createObjectURL(file);
+        },
+        [maxDimension, minDimension, onFileError, onFileSelect, onlySquare],
+    );
+
+    const { getRootProps, getInputProps } = useDropzone({
+        accept: acceptedFileTypes,
+        maxSize: maxFileSize,
+        disabled: isDisabled ?? currentState === InputFileAvatarState.SELECTING,
+        onDrop,
+        multiple: false,
+    });
+
+    const handleCancel = (event: React.MouseEvent<HTMLButtonElement>) => {
+        event.stopPropagation();
+        setImagePreview(null);
+        setCurrentState(InputFileAvatarState.IDLE);
+        if (imagePreview) {
+            URL.revokeObjectURL(imagePreview);
+        }
+    };
+
+    const overrideCheck = isDisabled ? InputFileAvatarState.DISABLED : variant ?? currentState;
+    const effectiveState = variantOverrideState[overrideCheck] ?? currentState;
+    const { containerClasses, addIconClasses } = stateToClassNames[effectiveState];
+
+    const inputAvatarClassNames = classNames(
+        'group flex size-16 items-center justify-center rounded-full bg-neutral-0 hover:shadow-neutral',
+        'focus:outline-none focus-visible:ring focus-visible:ring-offset',
+        containerClasses,
+    );
+
+    const iconClassNames = classNames(addIconClasses);
+
+    return (
+        <InputContainer id={id} useCustomWrapper {...otherContainerProps}>
+            <div {...getRootProps()} className={inputAvatarClassNames} aria-label="Select File">
+                <input {...getInputProps()} id={id} type="file" aria-label="Avatar Image Select" />
```

Comment:
The label will be set by the parent element
```suggestion
            <div {...getRootProps()} className={inputAvatarClassNames}>
                <input {...getInputProps()} id={id} type="file" />
```